tests/src/test/java/io/streamnative/pulsar/handlers/kop/KafkaApisTest.java

```diff
@@ -803,7 +812,89 @@ public void testBrokerRespectsPartitionsOrderAndSizeLimits() throws Exception {
             maxPartitionBytes, maxResponseBytes, messagesPerPartition);
     }
 
+    @DataProvider(name = "allowAutoTopicCreation")
+    public static Object[][] allowAutoTopicCreation() {
+        return new Object[][]{
+                { true, true, true },
+                { true, true, false},
+                { true, false, true },
+                { true, false, false},
+                { true, null, true },
+                { true, null, false},
+                { false, true, true },
+                { false, true, false},
+                { false, false, true },
+                { false, false, false},
+                { false, null, true },
+                { false, null, false}
+        };
+    }
+
     // verify Metadata request handling.
+    @Test(timeOut = 20000, dataProvider = "allowAutoTopicCreation")
+    public void testBrokerHandleTopicMetadataRequestAllowAutoTopicCreation(boolean brokerAllowAutoTopicCreation,
+                                                     Boolean overrideNameSpaceAutoTopicCreation,
+                                                     boolean allowAutoTopicCreationInRequest) throws Exception {
```

Comment:
Could you move this test into another test class (like `KafkaRequestHandlerTest`)? Currently the `KafkaApisTest`'s setup and cleanup methods are decorated with `@BeforeMethod` and `@AfterMethod`. It takes too much time to test.